source/server/connection_handler_impl.cc

```diff
@@ -44,19 +45,24 @@ void ConnectionHandlerImpl::addListener(absl::optional<uint64_t> overridden_list
     if (overridden_listener.has_value()) {
       for (auto& listener : listeners_) {
         if (listener.second.listener_->listenerTag() == overridden_listener) {
-          listener.second.tcp_listener_->get().updateListenerConfig(config);
+          listener.second.tcpListener()->get().updateListenerConfig(config);
           return;
         }
       }
       NOT_REACHED_GCOVR_EXCL_LINE;
     }
     auto tcp_listener = std::make_unique<ActiveTcpListener>(*this, config);
-    details.tcp_listener_ = *tcp_listener;
+    details.typed_listener_ = *tcp_listener;
     details.listener_ = std::move(tcp_listener);
   } else {
     ASSERT(config.udpListenerFactory() != nullptr, "UDP listener factory is not initialized.");
-    details.listener_ =
-        config.udpListenerFactory()->createActiveUdpListener(*this, dispatcher_, config);
+    ASSERT(worker_index_.has_value());
+    details.listener_ = config.udpListenerFactory()->createActiveUdpListener(*worker_index_, *this,
+                                                                             dispatcher_, config);
+    Network::UdpListenerCallbacks* udp_listener =
+        dynamic_cast<ActiveUdpListenerBase*>(details.listener_.get());
+    ASSERT(udp_listener != nullptr);
+    details.typed_listener_ = *static_cast<Network::UdpListenerCallbacks*>(udp_listener);
```

Comment:
Is this static cast needed? It's the type of the variable above?

Reply:
That must have been added during some previous variation. Removed.